lib/core/src/Cardano/Wallet.hs

```diff
@@ -759,10 +759,11 @@ newWalletLayer tracer bp db nw tl = do
                 ErrListTransactionsStartTimeLaterThanEndTime $
                     ErrStartTimeLaterThanEndTime start end
             _ -> pure ()
+        let slotIdRange = (error "TODO :: UTCTime -> SlotId") timeRange
```

Comment:
Re 
```
case (mStart, mEnd) of
            (Just start, Just end) -> when (start > end) $ throwE $
	                ErrListTransactionsStartTimeLaterThanEndTime $
	                    ErrStartTimeLaterThanEndTime start end
	            _ -> pure ()
```

I considered moving this logic to `Primitive.Types` as `isValid :: Range a -> Bool`, but then there is no nice way of constructing `ErrStartTimeLaterThanEndTime` with pure `UTCTime`s.